ivy/functional/frontends/torch/tensor.py

```diff
@@ -23,61 +23,61 @@ def add(self, other, *, alpha=1):
         return torch_frontend.add(self.data, other, alpha=alpha)
 
     def add_(self, other, *, alpha=1):
-        self.data = self.add(other, alpha=alpha)
+        self.data = self.add(other, alpha=alpha).data
         return self.data
 
     def asin(self):
         return torch_frontend.asin(self.data)
 
     def asin_(self):
-        self.data = self.asin()
+        self.data = self.asin().data
         return self.data
```

Comment:
Why exactly do these return `self.data`? Should they not return `self`? The current implementation will return an `ivy.Array` which is wrong I believe. Can you change these to all return `self` instead?